operator/iamkeycontroller/create.go

```diff
@@ -0,0 +1,149 @@
+package iamkeycontroller
+
+import (
+	"context"
+	"fmt"
+	pipeline "github.com/ccremer/go-command-pipeline"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	v2 "github.com/exoscale/egoscale/v2"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"github.com/vshn/provider-exoscale/operator/steps"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+)
+
+// Create implements managed.ExternalClient.
+func (p *IAMKeyPipeline) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	log := controllerruntime.LoggerFrom(ctx)
+	log.Info("Creating resource")
+
+	iam := fromManaged(mg)
+	if iam.Status.AtProvider.KeyID != "" {
+		// IAMKey already exists
+		return managed.ExternalCreation{}, nil
+	}
+
+	pipe := pipeline.NewPipeline().WithBeforeHooks(steps.DebugLogger(ctx)).
+		WithSteps(
+			pipeline.NewStepFromFunc("create IAM key", p.createIAMKeyFn(iam)),
+			pipeline.If(hasSecretRef(iam),
+				pipeline.NewStepFromFunc("create credentials secret", p.createCredentialsSecretFn(iam)),
+			),
```

Comment:
As argued before, we always need to create the secret
```suggestion
			pipeline.NewStepFromFunc("create credentials secret", p.createCredentialsSecretFn(iam)),
```